hydromt/data_adapter/geodataset.py

```diff
@@ -400,7 +420,12 @@ def _slice_spatial_dimension(
         logger.debug(f"Clip {predicate} [{bbox_str}] (EPSG:{epsg})")
         ds = ds.vector.clip_geom(geom, predicate=predicate)
         if ds.vector.index.size == 0:
-            _exec_nodata_strat("No data within spatial domain.", handle_nodata, logger)
+            if handle_nodata == NoDataStrategy.RAISE:
+                _exec_nodata_strat(
+                    "No data within spatial domain.", handle_nodata, logger
+                )
+            elif handle_nodata == NoDataStrategy.IGNORE:
+                logger.warn("No data within spacial domain for")
```

Comment:
Typo (spacial vs spatial). Also, this warning message feels incomplete due to the "for". Where you planning to add any information here? Otherwise just add something like "No data within spatial domain. skipping..." (similar to above)